### DISCUSSION TITLE
Logging

### DISCUSSION THREAD

**Discussion by SGFGOV** [medusajs/medusa#3524] [Feature Requests]
Medusa logger, though it implements a level of detail, there are some other details that would make the logger a lot more meaningful

1. Can we include timestamps? This is very useful to debug sequences of actions, particularly events. 
2. Can we include an identifier that defines where the log was generated such __dirname or __filename . Right now we need to manually search thecode base
3. Can we the logging be event-driven, and push events via an event handler like redis, and to other cloud agents like aws cloud watch. 
4. Can there be more graceful error handling in the logger? Currently if there an error in the parameter passed to the logger, the stack crashes. 

Ideally, I like something like this
```
[Timestamp][logleve][instance][identitifer] - [log message including optional parameter ] Jus makes it easier to filter in or process in spreadsheet like excel with clear delimiters
```
This maybe colour coded or even encoded as JSON.

**Comment by nolawnchairs** [comment]
Better yet, add an interface for people to add their own logger?
